doc/reference/configuration/configuration_reference.rst

```diff
@@ -3599,6 +3599,165 @@ The ``replication`` section defines configuration parameters related to :ref:`re
     | Default: ``false``
     | Environment variable: TT_REPLICATION_ANON
 
+.. _configuration_reference_replication_autoexpel:
+
+.. confval:: replication.autoexpel
+
+    The ``replication.autoexpel`` option designed for managing dynamic clusters using YAML-based configurations.
+    It enables the automatic expulsion of instances that are removed from the YAML configuration.
+
+    Only instances with names that match the specified prefix are considered for expulsion; all others are excluded.
+    Additionally, instances without a persistent name are ignored.
+
+    If an instance is in read-write mode and has the latest database schema, it initiates the expulsion of instances that:
+
+	- Match the specified prefix
+	- Absent from the updated YAML configuration
+
+    The expulsion process follows the standard procedure, involving the removal of the instance from the ``_cluster`` system space.
+
+    The ``autoexpel`` logic is activated during specific events:
+
+	- **Startup**. When the cluster starts, ``autoexpel`` checks and removes instances not matching the updated configuration.
+	- **Reconfiguration**. When the YAML configuration is reloaded, ``autoexpel`` compares the current state to the updated configuration and performs necessary expulsions.
+	- ``box.status`` **Watcher event**. Changes detected by the ``box.status watcher`` also trigger the ``autoexpel`` mechanism.
```

Comment:
Maybe even this is too detailed for the API reference alike article. Not sure, this is to be discussed.

Reply:
cc @p7nov

Reply:
Agree.